initialization/initialization_test.go

```diff
@@ -41,425 +46,466 @@ func (l testLogger) Debug(msg string, args ...interface{}) { l.t.Logf("\tDEBUG\t
 
 func TestInitialize(t *testing.T) {
 	r := require.New(t)
-	log := testLogger{t: t}
-
-	cfg, opts := getTestConfig(t)
-	commitment := make([]byte, 32)
-	init, err := NewInitializer(cfg, opts, commitment)
+	init, err := NewInitializer(getTestOptions(t)...)
 	r.NoError(err)
-	init.SetLogger(log)
 
-	var eg errgroup.Group
-	eg.Go(assertNumLabelsWrittenChan(init, t))
-	r.NoError(init.Initialize())
-	r.NoError(eg.Wait())
+	{
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
 
-	// Cleanup.
-	r.NoError(init.Reset())
+		var eg errgroup.Group
+		eg.Go(assertNumLabelsWritten(ctx, t, init))
+		r.NoError(init.Initialize(ctx))
+		cancel()
+		eg.Wait()
+	}
 }
```

Comment:
I don't understand this test. How is it verifying that Initializer actually does something? The checks in `assertNumLabelsWritten()` only check if `init.SessionNumLabelsWritten()` returns values that are monotonically nondecreasing. The initializer could just hang and do nothing and return 0 all the time. Am I missing something?

Reply:
The call to `init.Initialize()` is synchronous. If it returns without an error, the Initializer has completed it's work and calling `init.Status()` would return `StatusCompleted` and `init.SessionNumLabelsWritten()` would return `opts.NumUnits * cfg.LabelsPerUnit`. The later is the condition that is checked for Initializer to stop its work and the former returns `StatusCompleted` if that condition is met. Both of these conditions are tested for in other tests and are not the focus of this test.

If fewer labels would have been written but Initializer returns early it would return with an error.

Reply:
I still don't understand the merit of this test. It's called `TestInitialize()` yet it passes if I replace the body of the `Initialize()` method with `return nil`.

Reply:
I added an additional assert that checks if the expected number of labels have been created.